pybamm/solvers/c_solvers/idaklu/CasadiSolverOpenMP.cpp

```diff
@@ -125,36 +212,45 @@ void CasadiSolverOpenMP::SetMatrix() {
 }
 
 void CasadiSolverOpenMP::Initialize() {
+  int flag;
   // Call after setting the solver
 
   // attach the linear solver
-  if (LS == nullptr)
+  if (LS == nullptr) {
     throw std::invalid_argument("Linear solver not set");
-  IDASetLinearSolver(ida_mem, LS, J);
+  }
+  flag = IDASetLinearSolver(ida_mem, LS, J);
+  CheckErrors(flag);
```

Comment:
Not completely necessary since it is most likely optimized out, but you can simplify this since the flag is not needed later
```suggestion
  CheckErrors(IDASetLinearSolver(ida_mem, LS, J));
```

Reply:
Feel free to just make a cleanup ticket so this PR can advance. This is not necessary for this PR to be approved